aws-crt-kotlin/native/src/aws/sdk/kotlin/crt/Logging.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package aws.sdk.kotlin.crt
+
+import kotlinx.atomicfu.atomic
+import kotlinx.cinterop.*
+import libcrt.*
+
+@OptIn(ExperimentalForeignApi::class)
+internal object Logging {
+    private val initialized = atomic(false)
+
+    internal fun initialize(config: Config) {
+        if (!initialized.compareAndSet(false, true)) return
+
+        memScoped {
+            val options = cValue<aws_logger_standard_options> {
+                when (config.logDestination) {
+                    LogDestination.None -> return
+                    LogDestination.Stdout -> file = platform.posix.stdout
+                    LogDestination.Stderr -> file = platform.posix.stderr
+                    LogDestination.File -> { filename = config.logFile?.cstr?.ptr
+                        ?: throw IllegalArgumentException("LogDestination.File configured without logFile")
+                    }
```

Comment:
**Nit**: Unnecessary braces and `throw IllegalArgumentException` could be replaced with `requireNotNull`:

```kotlin
LogDestination.File -> filename = requireNotNull(config.logFile?.cstr?.ptr) { "LogDestination.File configured without logFile" }
```